presto-memory/src/main/java/io/prestosql/plugin/memory/MemorySplitManager.java

```diff
@@ -65,6 +68,13 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transactionHand
                 splits.add(new MemorySplit(table.getId(), i, splitsPerNode, dataFragment.getHostAddress(), rows, OptionalLong.empty()));
             }
         }
+
+        if (table.getSampleRatio().isPresent()) {
```

Comment:
This still has a semantic issue. It treats the following queries as equivalent:

```sql
SELECT *
FROM t TABLESAMPLE SYSTEM (50) 
LIMIT 1
```

vs

```sql
SELECT * FROM (
    SELECT *
    FROM t 
    LIMIT 1
)
TABLESAMPLE SYSTEM (50) 
```

For a table with a large number of rows, the first query should always return one row, while the second one should return 1 or 0 rows with a ~50% chance. So, the order of application of sample vs limit matters, but the current code doesn't take that into account.